### ISSUE TITLE
Adding 'curiosity' throws an error: ‘utils.py:309: UserWarning: This overload of nonzero is deprecated’

### ISSUE THREAD

**Issue by Eelam** [Unity-Technologies/ml-agents#5793]
**Describe the bug**
I added the Curiosity Intrinsic Reward to the Walker example and ran the mlagents-learn. It starts without any errors. Then it displays ‘‘site-packages\mlagents\trainers\torch\utils.py:309: UserWarning: This overload of nonzero is deprecated’.

The following lines were added to the Walker.YAML:
```
curiosity:
        strength: 0.1
        gamma: 0.995
        network_settings:
          hidden_units: 64
          learning_rate: 0.0001
```

Walker.YAML:
```
behaviors:
  Walker:
    trainer_type: ppo
    hyperparameters:
      batch_size: 2048
      buffer_size: 20480
      learning_rate: 0.0003
      beta: 0.005
      epsilon: 0.2
      lambd: 0.95
      num_epoch: 3
      learning_rate_schedule: linear
    network_settings:
      normalize: true
      hidden_units: 512
      num_layers: 3
      vis_encode_type: simple
    reward_signals:
      extrinsic:
        gamma: 0.995
        strength: 1.0
      curiosity:
        strength: 0.1
        gamma: 0.995
        network_settings:
          hidden_units: 64
          learning_rate: 0.0001
    keep_checkpoints: 5
    max_steps: 30000000
    time_horizon: 1000
    summary_freq: 10000
```

If I ran without the Curiosity Intrinsic Reward, there was no error.

**Screenshots**
I have added the screenshot of the console log. I have noticed that when I set the summary_freq: 100000, the error displayed before the first Step. Also, I tried without network_settings, and the error is still there.

**Environment (please complete the following information):**
- Unity Version: Unity 2020.3.33f1 & Unity 2021.3.10f1
- OS + version: Windows 11
- _ML-Agents version_: ML-Agents v0.28.0
- _Torch version_: 1.7.1+cu110
- _Environment_: Python 3.7.9
![MLAgents_curiosity](https://user-images.githubusercontent.com/7978145/193475519-84d1661d-6e52-495f-a887-a561603791a1.jpg)

**Comment by miguelalonsojr** [comment]
Hi! Thanks for reaching out. Unfortunately, we're unable to reproduce this issue. A couple of things that you could try: a) update to the newer version of ML-Agents v0.29 and/or b) update to python 3.8.10.

**Comment by Eelam** [comment]
Same issue: I have tried using python 3.8.10 and ML-Agents v0.29.
Screenshots: I have added the screenshot of the console log.
Is this anything to do with GPU? My system has 1070.
Unity environment with package version 2.2.1-exp.1 and communication version 1.5.0

Environment (please complete the following information):

- Unity Version: Unity 2020.3.33f1
- OS + version: Windows 11
- ML-Agents version: ML-Agents v0.29.0
- Torch version: 1.7.1+cu110
- Environment: Python 3.8.10

![Screenshot 2022-10-14 000440](https://user-images.githubusercontent.com/7978145/195728150-df9d100f-f97a-41df-85b4-4bcc0ed8353a.jpg)

**Comment by Eelam** [comment]
I fixed it. It was the CUDA tool's compatibility issue.
